genetic_algorithm/basic_string.py

```diff
@@ -0,0 +1,148 @@
+"""
+Simple multithreaded algorithm to show how the 4 phases of a genetic
+algorithm works (Evaluation, Selection, Crossover and Mutation)
+
+https://en.wikipedia.org/wiki/Genetic_algorithm
+
+
+Author: D4rkia
+"""
+
+from typing import List, Tuple
+import random
+
+N_POPULATION = 200
+N_SELECTED = 50
+MUTATION_PROBABILITY = 0.1
+random.seed(random.randint(0, 1000))
+
+
+def basic(sentence: str, genes: List[str]) -> Tuple[int, int, str]:
+    """The actual algorithm"""
+    # Verify that the sentence contains no other genes
+    # that the ones inside genes variable
+    for i in sentence:
+        if i not in genes_list:
+            raise ValueError(f"{i} is not in  genes list, evolution can't converge")
```

Comment:
```suggestion
    if any(c not in genes_list for c in sentence):
        raise ValueError(f"{c} is not in genes list, evolution can't converge")
```
More readable to use `c` for characters and `i` for integers.

Reply:
This gives me error with flake8 so it fail automated tests.

`F821 undefined name 'c'`